AFMReader/jpk.py

```diff
@@ -35,7 +35,62 @@ def _jpk_pixel_to_nm_scaling(tiff_page: tifffile.tifffile.TiffPage) -> float:
     return px_to_nm * 1e9
 
 
-# pylint: disable=too-many-locals
+def _get_z_scaling(tif: tifffile.tifffile, channel_idx: int) -> tuple[float, float]:
+    """
+    Extract the z scaling factor and offset for a JPK image channel.
+
+    Determined using JPKImageSpec.txt Version: 2.0:3fffffffffff supplied with JPK instrument.
+
+    Parameters
+    ----------
+    tif : tifffile.tifffile
+        A tiff file of .jpk images.
+
+    Returns
+    -------
+    tuple
+        A tuple contains values used to scale and offset raw data.
+    """
+    NrOfSlots = tif.pages[channel_idx].tags["32896"].value
+    default_slot = tif.pages[channel_idx].tags["32897"]
+
+    # Create a dictionary of list for the differnt slots
+    slots = {}
+    for slot in range(NrOfSlots):
+        slots[slot + 1] = []
+
+    # Extract the tags with numerical names in each slot
+    while NrOfSlots >= 1:
+        for tag in tif.pages[channel_idx].tags:
+            try:
+                tag_name_float = float(tag.name)
+                if tag_name_float >= (32912 + ((NrOfSlots - 1) * 48)) and tag_name_float < (32912 + (NrOfSlots * 48)):
+                    slots[NrOfSlots].append(tag.name)
+            except ValueError:
+                continue
+        NrOfSlots -= 1
+
+    # Find the number of the default slot (selected in the instrument GUI)
+    for slot_name, values in slots.items():
+        for value in values:
+            if tif.pages[channel_idx].tags[str(value)].value == default_slot.value:
+                default_slot_number = slot_name
```

Comment:
Similarly I think plain `default_slot` would do here (and where it is referenced below).

```suggestion
                default_slot = slot_name
```

Reply:
I'm just adverse to using suffixes which describe what something is e.g. `_dict` , `_list`, `_number`, `_str` when type hinting and context exists to indicate the type of something (but then Python is unfortuantely a dynamically typed language). Knowing that its a `_number` doesn't tell the reader of the code anything extra (notionally its mostly irrelevant what type it is).

One approach I use in such instances is to use a prefix of `_` as this indicates variables that the variable is private and not accessible outside of the function. Another approach is a suffix of `_` and this is used when there are conflicts with Python keywords and built-ins.